pkg/threescale/backend/manager.go

```diff
@@ -220,58 +220,85 @@ func (b *Backend) Flush() {
 		}
 
 		for _, cachedApp := range service {
+			if cachedApp.reportingErr && cachedApp.authErr {
+				// nothing to do here only continue on
+				// todo add logging
+				continue
+			}
+
+			//now that we have handled the dual error case, we know we need to
+			// read from the cache and modify the entry so we take a write lock
 			app, ok := b.cache.Get(cachedApp.cacheKey)
 			if !ok {
 				continue
 			}
+
 			app.Lock()
+			fetchedState := b.applicationFromResponse(cachedApp.authResp)
+
+			// we know auth has been successful here or we wouldn't be here
```

Comment:
Why? It is not obvious to me as a reader, and in fact there is an `if` clause below testing for an auth error.

Reply:
we have four cases to cover
 1. Report error and auth error - we check this at top and fail early
2. Report error and auth success (as you highlighted) - so it makes sense here that authErr must be false
3. Report success and auth error
4. Report success and auth success

They are from top to bottom handled here. I can make the ifs more explicit if you think that it reads better? I actually debated not doing it at the time with myself

Reply:
fixed for clarity